sharding/nodesSetup.go

```diff
@@ -160,57 +241,64 @@ func (ns *NodesSetup) processConfig() error {
 		ns.nrOfNodes++
 	}
 
-	if ns.ConsensusGroupSize < 1 {
-		return ErrNegativeOrZeroConsensusGroupSize
-	}
-	if ns.MinNodesPerShard < ns.ConsensusGroupSize {
-		return ErrMinNodesPerShardSmallerThanConsensusSize
-	}
-	if ns.nrOfNodes < ns.MinNodesPerShard {
-		return ErrNodesSizeSmallerThanMinNoOfNodes
+	for _, consensusConfig := range ns.chainParametersByEpoch {
```

Comment:
:+1: good, we check all provided values on the constructor, ensuring early signaling of a bad config.